backend/infrahub/api/transformation.py

```diff
@@ -43,13 +49,21 @@ async def transform_python(
     transform = await NodeManager.get_one_by_id_or_default_filter(
         db=db,
         id=transform_id,
-        kind=InfrahubKind.TRANSFORMPYTHON,
+        kind=CoreTransformPython,
         branch=branch_params.branch,
         at=branch_params.at,
     )
 
-    query = await transform.query.get_peer(db=db)  # type: ignore[attr-defined]
-    repository = await transform.repository.get_peer(db=db)  # type: ignore[attr-defined]
+    query = await transform.query.get_peer(db=db, peer_type=CoreGraphQLQuery, raise_on_error=True)
+    repository = await transform.repository.get_peer(db=db, peer_type=CoreGenericRepository, raise_on_error=True)
+
+    if repository.commit.value is None:
```

Comment:
This line will still be problematic as the CoreGenericRepository type doesn't have a `commit` attribute. Only CoreRepository and CoreReadOnlyRepository has that. A simple workaround in this example could be to just use CoreRepository, or leave this line as `# type: ignore[attr-defined]` for now as it's not really related to the overall work with this PR.